pkg/controller/eventing/inmemory/channel/reconcile.go

```diff
@@ -149,19 +149,34 @@ func (r *reconciler) reconcile(ctx context.Context, c *eventingv1alpha1.Channel)
 
 	util.AddFinalizer(c, finalizerName)
 
-	svc, err := util.CreateK8sService(ctx, r.client, c)
+	cCopy := c.DeepCopy()
+	cCopy.Spec.Provisioner.Name = asyncProvisionerName
+	svc, err := util.CreateK8sService(ctx, r.client, cCopy)
```

Comment:
Why are we always using `in-memory` here? Is it only because of my comment in the tests?